nightingale_navigation/hallway_planner/include/hallway_planner/hallway_planner.hpp

```diff
@@ -0,0 +1,135 @@
+#pragma once                                     
+/*********************************************************************
+ * Software License Agreement (MIT License)
+ * Author: Younes Reda
+ *********************************************************************/
+
+
+#include <ros/ros.h>
+#include <costmap_2d/costmap_2d_ros.h>
+#include <geometry_msgs/Pose.h>
+#include <global_planner/planner_core.h>
+#include <vector>
+#include <opencv2/opencv.hpp>
+#include <optional>
+#include <string>
+#include <mutex>
+#include <costmap_2d/cost_values.h>
+#include <hallway_planner/hallway_locator.hpp>
+
+namespace hallway_planner
+{
+
+class HallwayPlanner : public global_planner::GlobalPlanner
+{
+public:
+
+    /**
+     * @brief  Default con/destructor for the PlannerCore object
+     */
+    HallwayPlanner() : global_planner::GlobalPlanner(), hallway_map_() {};
+    ~HallwayPlanner() = default;
+
+
+    /**
+     * @brief Copies
+     */
+    HallwayPlanner(const HallwayPlanner &) = default;
+    HallwayPlanner &operator=(const HallwayPlanner &)=default;
+    
+
+    /**
+     * @brief Moves
+     */
+    HallwayPlanner(HallwayPlanner &&) = default;
+    HallwayPlanner &operator=(HallwayPlanner &&) = default;
+
+    /**                                                                                                                                                                                                  
+       * @brief Given a goal pose in the world, compute a plan
+       * @param start The start pose 
+       * @param goal The goal pose 
+       * @param plan The plan... filled by the planner
+       * @return True if a valid plan was found, false otherwise
+       */
+    bool makePlan(const geometry_msgs::PoseStamped& start, 
+                  const geometry_msgs::PoseStamped& goal, std::vector<geometry_msgs::PoseStamped>& plan) override
+    {
+      if (global_planner::GlobalPlanner::makePlan(start,goal,plan))
+      {
+        // younes todo do hallway biasing
+        return true;
+      }
+      return false;
+    }
+
+    /**
+       * @brief Given a goal pose in the world, compute a plan
+       * @param start The start pose 
+       * @param goal The goal pose 
+       * @param plan The plan... filled by the planner
+       * @param cost The plans calculated cost
+       * @return True if a valid plan was found, false otherwise
+       */
+      bool makePlan(const geometry_msgs::PoseStamped& start, 
+                    const geometry_msgs::PoseStamped& goal, std::vector<geometry_msgs::PoseStamped>& plan,
+                    double& cost) override
+      {
+        cost = -1;
+        if (hallway_planner::HallwayPlanner::makePlan(start,goal,plan))
```

Comment:
how will the plan be made? Will you be running an existing version of A* on the costmap you generated from the hallway locator?

Reply:
the original plan will be made through using the old GlobalPlanner (the default globalplanner that ROS provides) which does A* or djikstra depending how u config. i think we config djikstra. Then the hallways are detected and the path is biased